### PR TITLE
Add documentation for protected content in the Delivery API

### PR DESCRIPTION
**Pull request by kjac**
## Description

Added docs for the upcoming "protected content" feature in the Delivery API. The feature is slated for V12.3.

⚠️ Do NOT merge this in until V12.3RC is out ⚠️ 

## Type of suggestion

* [ ] Typo/grammar fix
* [x] Updated outdated content
* [x] New content
* [ ] Updates related to a new version
* [ ] Other

## Product & version (if relevant)

V12.3

## Deadline (if relevant)

V12.3RC is out on October 19, so it would be nice for this to go out then too.

### CODE REVIEW

**Review by elit0451 — APPROVED** [review]
Great docs! 💪

File: 12/umbraco-cms/reference/content-delivery-api/protected-content-in-the-delivery-api.md [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+    services.ConfigureOptions<Umbraco.Cms.Api.Delivery.Configuration.ConfigureUmbracoMemberAuthenticationDeliveryApiSwaggerGenOptions>();
+
+    // ...
+}
+```
+{% endcode %}
+
+The Swagger UI will now feature authorization.
+
+Remember to use `umbraco-member` as `client_id` when authorizing. `client_secret` can be omitted, as it is not used by the authorization flow.
```
[the remark this answers is not in the document]
Comment (kjac) [reply 12/umbraco-cms/reference/content-delivery-api/protected-content-in-the-delivery-api.md:388]: Let's do that. 👍 will amend

File: 12/umbraco-cms/reference/content-delivery-api/protected-content-in-the-delivery-api.md [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+- `code_challenge=WZRHGrsBESr8wYFZ9sx0tPURuZgG2lmzyvWpwXPKz8U`
+- `code_challenge_method=S256`
+
+If everything works as expected, the request will yield a redirect to the login page. Completing the login form will cause a redirect to the specified redirect URL with a `code` query string parameter. The `code` can subsequently be exchanged for an access token, which can be used to access protected content.
+
+{% hint style="info" %}
+Do not worry about the URL construction and subsequent handling of the `code` parameter. This complexity is what the OpenId Connect client libraries handle for us.
+{% endhint %}
+
+For more inspiration on using the built-in member authentication, visit [this tutorial](https://docs.umbraco.com/umbraco-cms/tutorials/members-registration-and-login). Here you will also learn how to create member sign-up functionality.
```
Comment (alina-tincas) [inline 12/umbraco-cms/reference/content-delivery-api/protected-content-in-the-delivery-api.md:212]:
```suggestion
For more inspiration on using the built-in member authentication, check the [Members Registration and Login](https://docs.umbraco.com/umbraco-cms/tutorials/members-registration-and-login) article. Here you will also learn how to create member sign-up functionality.
```

File: 12/umbraco-cms/reference/content-delivery-api/protected-content-in-the-delivery-api.md [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+{
+    services.AddUmbraco(_env, _config)
+        // ...
+        .AddGitHubAuthentication()
+        .Build();
+}
+```
+{% endcode %}
+
+Now we can test the setup. Again we'll be calling `https://{server-host}/umbraco/delivery/api/v1/security/member/authorize` as described previously, but we need to add one more query string parameter:
```
Comment (alina-tincas) [inline 12/umbraco-cms/reference/content-delivery-api/protected-content-in-the-delivery-api.md:304]: Again and previously seem to be the in the same sentence 🤔 
```suggestion
Now we can test the setup. We'll be calling `https://{server-host}/umbraco/delivery/api/v1/security/member/authorize` as described previously, but we need to add one more query string parameter:
```

**Review by alina-tincas — CHANGES_REQUESTED** [review]
Hi @kjac thank you for the PR 🙌 

Awesome to see it has been approved by @elit0451 as well :h5YR

I have just a few suggestions, can you please take a look at them and let me know what you think? 😊 

Then I will make sure it is merged in on on October 19 💪

File: 12/umbraco-cms/reference/content-delivery-api/protected-content-in-the-delivery-api.md [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+
+{% hint style="info" %}
+Remember to use `umbraco-member` as `client_id` when authorizing. `client_secret` can be omitted, as it is not used by the authorization flow.
+{% endhint %}
+
+## Client configuration samples
+
+The following samples show how to configure an ASP.NET Core client to utilize member authorization in the Delivery API.
+
+To put these samples into context, please refer to the article above.
```
[the remark this answers is not in the document]
Comment (kjac) [reply 12/umbraco-cms/reference/content-delivery-api/protected-content-in-the-delivery-api.md:396]: It's all the previous sections of this article. We have opted to move the samples to the very end of the article, to keep the flow as clear as possible. Does that make sense?

[the remark this answers is not in the document]
Comment (alina-tincas) [reply 12/umbraco-cms/reference/content-delivery-api/protected-content-in-the-delivery-api.md:396]: ah I see now! Then all is good 👍

**Review by alina-tincas — APPROVED** [review]

**Pull request merged**